WcaOnRails/app/models/post.rb

```diff
@@ -2,11 +2,37 @@
 
 class Post < ApplicationRecord
   belongs_to :author, class_name: "User"
+  has_many :post_tags, autosave: true, dependent: :destroy
 
   validates :title, presence: true, uniqueness: true
   validates :body, presence: true
   validates :slug, presence: true, uniqueness: true
 
+  def reload
+    @tags = nil
+    super
+  end
+
+  attr_writer :tags
+
+  def tags
+    @tags ||= post_tags.pluck(:tag).join(",")
```

Comment:
I have some concerns with moving the code to the controller:

1. The model tests I wrote will no longer be applicable. Porting them to controller tests is doable, but controller tests are deprecated in Rails 5. Maybe they'd have to become request specs. This isn't a reason not to make the switch, but it's a reason why making this switch will be a bit of work, so I'd like to make sure we're sure before I got to much effort here.
2. simple_form will no longer "just work". I suspect this will balloon out into more code than you might immediately suspect:
  - For example: since this logic is no longer in the model, validations will not longer "just work", which means we'll have to write code in the controller to pass validation errors down into the view, and the view code will have to be updated to check if these errors exist and conditionally generate more html to show them (all work that is normally handled for us by simple_form).

I may be suffering from "I just got this working, and I'm too lazy to change this" syndrome. If you think my fears are unfounded, would you be willing to try out your way and send in a PR? I think I'll merge this up as it is tomorrow, I don't think that prevents us from improving the code later on.